src/components/accession2/withdraw/WeightWithdrawal.tsx

```diff
@@ -0,0 +1,241 @@
+        onChangeWithdrawnQuantity(accession.remainingQuantity?.quantity);
+      }
+    } else {
+      setWithdrawnQuantity(undefined);
+    }
+
+    setWithdrawAllSelected(withdrawAll);
+  };
+
+  const validateAmount = (estimated: number, withdrawnQty: number, withdrawnUnits: string) => {
+    if (withdrawnQty) {
+      if (isNaN(withdrawnQty)) {
+        setWithdrawnQtyError(strings.INVALID_VALUE);
+        return false;
+      }
+      if (Number(withdrawnQty) <= 0) {
+        setWithdrawnQtyError(strings.INVALID_VALUE);
+        return false;
+      }
+      if (accession.remainingQuantity?.units && Number(withdrawnUnits === accession.remainingQuantity.units)) {
+        if (accession.remainingQuantity.units === 'Seeds') {
+          if (Number(estimated) > accession.remainingQuantity?.quantity) {
+            setWithdrawnQtyError(strings.WITHDRAWN_QUANTITY_ERROR);
+            return false;
+          }
+        } else if (withdrawnQty > accession.remainingQuantity?.quantity) {
+          setWithdrawnQtyError(strings.WITHDRAWN_QUANTITY_ERROR);
+          return false;
+        }
+      }
+      if (purpose === 'Nursery' || purpose === 'Viability Testing') {
+        if (!accession.estimatedCount || !accession.subsetWeight?.quantity || !accession.subsetCount) {
+          setWithdrawnQtyError(
+            purpose === 'Nursery'
+              ? strings.MISSING_SUBSET_WEIGHT_ERROR_NURSERY
+              : strings.MISSING_SUBSET_WEIGHT_ERROR_VIABILITY_TEST
+          );
+          return false;
+        }
+      }
+      setWithdrawnQtyError('');
+      return true;
+    } else {
+      setWithdrawnQtyError(strings.REQUIRED_FIELD);
+      return false;
+    }
+  };
+
+  return (
+    <Grid container direction='row'>
+      {accession.subsetWeight?.quantity && accession.subsetCount ? (
+        <Grid container item xs={12}>
+          <Grid item xs={6} sx={{ marginTop: theme.spacing(2), marginBottom: theme.spacing(2) }}>
+            <Textfield
+              label={strings.SUBSET_WEIGHT}
+              id='subsetWeight'
+              type='number'
+              value={`${accession.subsetWeight?.quantity} ${accession.subsetWeight?.units}`}
+              display={true}
+              tooltipTitle={strings.SUBSET_WEIGHT_REQUIRED}
+            />
+          </Grid>
+          <Grid item xs={6} sx={{ marginTop: theme.spacing(2), marginBottom: theme.spacing(2) }}>
+            <Textfield
+              label={strings.SUBSET_COUNT}
+              id='subsetCount'
+              type='number'
+              value={accession.subsetCount}
+              display={true}
+            />
+          </Grid>
+        </Grid>
+      ) : null}
+      <Grid container direction='row' justifyContent='space-between'>
+        <Grid item xs={accession.subsetWeight?.quantity && accession.subsetCount ? 6 : 12} paddingBottom={2}>
+          <Box display='flex' alignItems={withdrawnQtyError ? 'center' : 'end'}>
+            <Textfield
+              label={strings
```

Comment:
This section might benefit from being turned into its own function.

Reply:
Opted against this.  I didn't feel like it was worth the extra information passing and duplicated code.